bofire/surrogates/mlp.py

```diff
@@ -109,40 +125,48 @@ def forward(self, X: Tensor):
     @property
     def num_outputs(self) -> int:
         r"""The number of outputs of the model."""
-        return self.mlps[0].layers[-1].out_features  # type: ignore
+        return self.mlps[0].layers[-2].out_features  # type: ignore
 
 
 def fit_mlp(
     mlp: MLP,
-    dataset: RegressionDataSet,
+    dataset: MLPDataset,
     batch_size: int = 10,
     n_epoches: int = 200,
     lr: float = 1e-4,
     shuffle: bool = True,
     weight_decay: float = 0.0,
+    loss_function=nn.L1Loss,  # type: ignore
```

Comment:
Should we somehow list and check which we support currently?

Reply:
How do you recommend?

Reply:
Something like Union[torch.nn.modules.loss.L1Loss, ...]?

Reply:
Let's forget it. It is fine.